### PR TITLE
[NetAppFiles] bump to 2024-03-01-preview

### PR DESCRIPTION
**Pull request by audunn**
Migrate from msrestazure.tools to azure.mgmt.core.tools
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
NetAppFiles

File: src/netappfiles-preview/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.5.2
```
Comment (AllyW) [inline src/netappfiles-preview/HISTORY.rst:5]: version here should be consistent with setup.py, which is 1.0.0b2

Comment (audunn) [reply src/netappfiles-preview/HISTORY.rst]: Fixed

**Comment by audunn** [comment]
@AllyW updated

**Comment by audunn** [comment]
aaz  https://github.com/Azure/aaz/pull/560

**Review by kairu-ms — APPROVED** [review]

**Comment by kairu-ms** [comment]
@audunn, please verify the above breaking change are expected and write then in the release note. Thanks.

**Comment by audunn** [comment]
@kairu-ms  This should also satisfy the requrement to drop msrestazure dependency from CLI extensions.  Is there specific confirmation of this needed ? 

https://microsoft.sharepoint.com/:x:/t/AzureCLITools237/ETrztIl6PXhMpt22vr_5aSoBl7vdAq8lXRocPJhelkUPxA?e=ETZxaJ&CID=BE6A6F46-D970-4795-BE7C-B8692DB941ED&wdLOR=cA301C305-6C4B-43FB-B162-8B01A95B8FAE

File: src/netappfiles-preview/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,18 @@
 
 Release History
 ===============
+1.0.0b2
```
Comment (jsntcy) [inline src/netappfiles-preview/HISTORY.rst:5]: It seems there are no release notes for `1.0.0b1`.

Comment (audunn) [reply src/netappfiles-preview/HISTORY.rst:5]: You mean thye previous version? Do we need to add notes for that  here?

**Review by kairu-ms — APPROVED** [review]

File: src/netappfiles-preview/HISTORY.rst (hunk 2 above)
Comment (kairu-ms) [reply src/netappfiles-preview/HISTORY.rst:5]: Please add the history for 1.0.0b1 in later PRs

**Pull request merged**